### PR TITLE
Add --remote_header support

### PR DESCRIPTION
**Pull request by brentleyjones**

### CODE REVIEW

File: src/main/java/com/google/devtools/build/remote/client/RemoteOptions.java [hunk 1]
```diff
@@ -42,4 +45,9 @@
       names = "--remote_instance_name",
       description = "Value to pass as instance_name in the remote execution API.")
   public String remoteInstanceName = "";
+
+  @DynamicParameter(
+      names = "--remote_header",
+      description = "Dynamic parameters go here")
```
Comment (brentleyjones) [inline src/main/java/com/google/devtools/build/remote/client/RemoteOptions.java:51]: I'll fix

**Comment by brentleyjones** [comment]
@eytankidron @werkt Could I get someone to look at this please? Thanks!

File: src/main/java/com/google/devtools/build/remote/client/GrpcRemoteCache.java [hunk 2]
```diff
@@ -66,16 +70,33 @@ public static boolean isRemoteCacheOptions(RemoteOptions options) {
...
   }
 
+  private static ClientInterceptor customHeadersInterceptor(Map<String, String> headers) {
+    Metadata metadata = new Metadata();
+    for (Map.Entry<String,String> entry : headers.entrySet()) {
+      metadata.put(
+        Metadata.Key.of(entry.getKey(), Metadata.ASCII_STRING_MARSHALLER),
+        entry.getValue()
+      );
+    }
```
Comment (werkt) [inline src/main/java/com/google/devtools/build/remote/client/GrpcRemoteCache.java:80]: Two things spring to mind here: encodings, and the repetition of header key entries (multiple of the same key) - I was pretty sure that grpc/http2 headers matched with http1, where repetitions of keys was accepted.

Might want to try that with some non-ascii compatible UTF8 or some such, to see the result.

Comment (brentleyjones) [reply src/main/java/com/google/devtools/build/remote/client/GrpcRemoteCache.java:80]: Yeah, repetition should append. I'll work on both.

**Comment by werkt** [comment]
@brentleyjones ping?

**Comment by brentleyjones** [comment]
Thanks for the ping @werkt. Among other things I changed jobs, but that has settled down. I've added this back to my todo list.

File: src/main/java/com/google/devtools/build/remote/client/GrpcRemoteCache.java (hunk 2 above)
Comment (brentleyjones) [reply src/main/java/com/google/devtools/build/remote/client/GrpcRemoteCache.java:80]: Got encoding, bazel handles it way for this flag as well, so I think we are good there: https://github.com/bazelbuild/bazel/blob/4de6bcaa2d0489f91746c5700fbe99f49fa9d0c5/src/main/java/com/google/devtools/build/lib/remote/util/TracingMetadataUtils.java#L110-L124

Comment (brentleyjones) [reply src/main/java/com/google/devtools/build/remote/client/GrpcRemoteCache.java:80]: It seems it handles repetition with the same code as well.

Comment (brentleyjones) [reply src/main/java/com/google/devtools/build/remote/client/GrpcRemoteCache.java:80]: @werkt

**Pull request merged**